Use a DateRange Value Object to validate ranges

WouterSioen opened the pull request:
This makes sure we only can get valid date ranges injected, which makes
adding extra validation rules unnecessary and cleanly encapsulates the
date range logic.

WouterSioen committed:
Use a DateRange Value Object to validate ranges

This makes sure we only can get valid date ranges injected, which makes
adding extra validation rules unnecessary and cleanly encapsulates the
date range logic.

vanbrabantf commented:
I really like this. 
The only downside here is that an enduser has to use the `Daterange` class to call `getDownloadsBetweenDates`.

This is a good thing, as it get's early validated. But it does add a small hurdle to the user.

WouterSioen committed:
Add a seperate method to use the DateRange

This will remove the hurdle for end users to create a DateRange if they
don't want to.

WouterSioen commented:
I've updated to keep the `getDownloadsBetweenDates` backwards compatible but added a `getDownloadsInDateRange` method to make use of the new DateRange VO.

WouterSioen committed:
Avoid unneeded temporary variable.

vanbrabantf commented:
You are a diamond Wouter!
